File: False or Falsi Position method.m
%{
  Input 
      lower = 0.5
      upper = 1.0
      Error Accept <= 0.01 % 

    Output:
      0.5000    0.5980    0.5606    6.6615

      0.5000    0.5606    0.5573    0.5930

      0.5000    0.5573    0.5571    0.0513

      0.5000    0.5571    0.5570    0.0044
   
   Call Method: FalsePosition(0.5,1.0,0.01)
%}

function FalsePosition(lw, up, erRate)

     xr = callRoot(lw, up);
     preXr = xr;
     
     while(true)
         temp = callFunction(lw) * callFunction(xr);
         if(temp > 0) 
             lw = xr;
         elseif(temp < 0) 
             up = xr;
         else
             break;
         end
         
         xr = callRoot(lw, up);
         
         perCent = (abs(xr - preXr)/xr) * 100;
         output=[lw' up' xr' perCent'];
         disp(output);
         if(perCent < erRate) 
             break;
         end
         preXr = xr;
     end
end

function xr = callRoot(lw, up)
     xr = up - ((callFunction(up)*(lw - up))/(callFunction(lw) - callFunction(up)));
end

function funVal = callFunction(x)
    funVal = power(x, 5) - (8 * power(x, 4)) + (44 * power(x, 3)) - (91 * power(x, 2)) + (85 * x) - 26;
end

File: False or Falsi Position method.cpp
/*
  Input
    Low Position = 0.5 
    Upper Position = 1.0
    Error Accept = 0.1%
  Output
    0.5 0.59799 0.560643 6.66146
    0.5 0.560643 0.557338 0.592956
    0.5 0.557338 0.557052 0.051288
*/


#include <bits/stdc++.h>
using namespace std;

double callFunction(double x)
{
    double ans = pow(x, 5) - (8 * pow(x, 4)) + (44 * pow(x, 3)) - (91 * pow(x, 2));
    ans += (85 * x) - 26;
    return ans;
}

double callRoot(double lw, double up)
{
    double xr = up - ((callFunction(up)*(lw - up))/(callFunction(lw) - callFunction(up)));
    return xr;
}

int main() {

    double lw, up, xr;

    cin >> lw >> up;

    xr = callRoot(lw, up);

    double  preXr = xr;
    while (true) {
        double temp = callFunction(lw) * callFunction(xr);
        if(temp < 0) up = xr;
        else if(temp > 0) lw = xr;
        else break;

        xr = callRoot(lw, up);
        double perCent = (abs(xr - preXr)/xr) * 100;

        cout << lw << " " << up << " " << xr << " " << perCent << endl;

        if(perCent < 0.1) break;
        preXr = xr;
    }

    return 0;
}
